internal/services/netapp/netapp_volume_quota_rule_resource.go

```diff
@@ -0,0 +1,246 @@
+			}
+
+			// Performing some validations that are not possible in the schema
+			if errorList := netAppValidate.ValidateNetAppVolumeQuotaRule(ctx, pointer.From(volumeID), metadata.Client, pointer.To(model)); len(errorList) > 0 {
+				return fmt.Errorf("one or more issues found while performing deeper validations for %s:\n%+v", id, errorList)
+			}
+
+			parameters := volumequotarules.VolumeQuotaRule{
+				Location: location.Normalize(model.Location),
+				Properties: &volumequotarules.VolumeQuotaRulesProperties{
+					QuotaSizeInKiBs: pointer.To(model.QuotaSizeInKiB),
+					QuotaType:       pointer.To(volumequotarules.Type(model.QuotaType)),
+					QuotaTarget:     utils.String(model.QuotaTarget),
+				},
+			}
+
+			err = client.CreateThenPoll(ctx, id, parameters)
+			if err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+
+			return nil
+		},
+	}
+}
+
+func (r NetAppVolumeQuotaRuleResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 120 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.NetApp.VolumeQuotaRules
+
+			id, err := volumequotarules.ParseVolumeQuotaRuleID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			metadata.Logger.Infof("Decoding state for %s", id)
+			var state netAppModels.NetAppVolumeQuotaRuleModel
+			if err := metadata.Decode(&state); err != nil {
+				return err
+			}
+
+			metadata.Logger.Infof("Updating %s", id)
+
+			update := volumequotarules.VolumeQuotaRulePatch{
+				Properties: &volumequotarules.VolumeQuotaRulesProperties{},
+			}
+
+			update.Properties.QuotaSizeInKiBs = utils.Int64(state.QuotaSizeInKiB)
+			if err := client.UpdateThenPoll(ctx, pointer.From(id), update); err != nil {
+				return fmt.Errorf("updating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+
+			return nil
+		},
+	}
+}
+
+func (r NetAppVolumeQuotaRuleResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+
+			client := metadata.Client.NetApp.VolumeQuotaRules
+
+			id, err := volumequotarules.ParseVolumeQuotaRuleID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			metadata.Logger.Infof("Decoding state for %s", id)
+			var state netAppModels.NetAppVolumeQuotaRuleModel
+			if err := metadata.Decode(&state); err != nil {
+				return err
+			}
+
+			existing, err := client.Get(ctx, pointer.From(id))
+			if err != nil {
+				if existing.HttpResponse.StatusCode == http.StatusNotFound {
+					return metadata.MarkAsGone(id)
+				}
+				return fmt.Errorf("retrieving %s: %v", id, err)
+			}
+
+			volumeID := fmt.Sprintf("/subscriptions/%v/resourceGroups/%v/providers/Microsoft.NetApp/netAppAccounts/%v/capacityPools/%v/volumes/%v", id.SubscriptionId, id.ResourceGroupName, id.NetAppAccountName, id.CapacityPoolName, id.VolumeName)
```

Comment:
We should create a new VolumeID here instead through
```suggestion
			volumeID := volumequotarules.NewVolumeID(id.SubscriptionId, id.ResourceGroupName, id.NetAppAccountName, id.CapacityPoolName, id.VolumeName)
```

Reply:
I tried that but it don't convert to string, it tries to place the volume id struct, even if I use volumeID.String(), do you have an example on how to convert the struct into string?